File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/BlankItem.java
package com.spvessel.spacevil;

/**
 * BlankItem is pure subclass of Prototype without any extensions.
 * <p>
 * Example: used as cheap version of com.spvessel.spacevil.ButtonCore
 * (com.spvessel.spacevil.ButtonCore contains text and additional methods
 * extensions).
 */
public class BlankItem extends Prototype {
    static int count = 0;

    /**
     * Default BlankItem constructor.
     */
    public BlankItem() {
        setItemName("BlankItem_" + count);
        count++;
    }
}

File: SpaceVIL/Items/BlankItem.cs
namespace SpaceVIL
{
    /// <summary>
    /// BlankItem is pure subclass of Prototype without any extensions.
    /// <para/> Example: used as cheap version of SpaceVIL.ButtonCore 
    /// (SpaceVIL.ButtonCore contains text 
    /// and additional methods extensions).
    /// </summary>
    public class BlankItem : Prototype
    {
        static int count = 0;
        /// <summary>
        /// Default BlankItem constructor.
        /// </summary>
        public BlankItem()
        {
            SetItemName("BlankItem_" + count);
            count++;
        }
    }
}
